RevenueCatUI/Templates/Components/Packages/Package/PackageComponentView.swift

```diff
@@ -20,11 +20,18 @@ import SwiftUI
 @available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
 struct PackageComponentView: View {
 
+    @EnvironmentObject
+    private var paywallState: PaywallState
+
     let viewModel: PackageComponentViewModel
 
     var body: some View {
         // WIP: Do something with package id and selection
-        StackComponentView(viewModel: self.viewModel.stackComponentViewModel)
+        Button {
+            self.paywallState.select(package: self.viewModel.package)
```

Comment:
I know SwiftUI has different recommended patterns from Jetpack Compose, but in Compose land the use of `CompositionLocal`s ( = `@EnvironmentObject`s) is discouraged, because it introduces hidden dependencies to the views. As an alternative, this view could take an `onPackageSelected` closure, which gets called by the button. 

Not a blocker, but just wanted to mention it. 😇

Reply:
This is a completely fair question 😅 This `PaywallState` _could_ get passed in as both a `Binding<PaywallState>` and an `onPackageSelected` closure to the views. However, that kind of introduces a giant list of parameters that would need to get updated and passed into all these view recursively. I choose to to down the `@EnvironmentObject` route for the reason of simplifying the construction of all the views and any unknown callbacks or state that we know right now.

I _do_ agree that if this gets used too much and in a complex way that its unclear if that object will exist or not. However, in our case, the `PaywallState` gets injected _very_ much at the entry point of paywalls in `TemplateComponentsView` so any views are guaranteed to have it 🤷‍♂️

Reply:
Alright, yea this seems like a reasonable trade-off indeed!